src/auth.ts

```diff
@@ -0,0 +1,40 @@
+import { rule, shield } from "graphql-shield";
+import * as jwt from "jsonwebtoken";
+import { Context } from "./context";
+
+// App secret
+export const APP_SECRET = 'appsecret321'
+
+// Token
+interface Token {
+    userId: string
+}
+
+// Auth
+export function getUserId(context: Context) {
```

Comment:
I would suggest to move this to the context, so we can just lookup the userid on the context. In that way we can keep the authorization check much cleaner